insonmnia/miner/builder.go

```diff
@@ -139,7 +144,26 @@ func (b *MinerBuilder) Build() (miner *Miner, err error) {
 
 	log.G(ctx).Info("collected Hardware info", zap.Any("hardware", hardwareInfo))
 
-	grpcServer := grpc.NewServer(grpc.RPCCompressor(grpc.NewGZIPCompressor()), grpc.RPCDecompressor(grpc.NewGZIPDecompressor()))
+	var servOpts = []grpc.ServerOption{grpc.RPCCompressor(grpc.NewGZIPCompressor()), grpc.RPCDecompressor(grpc.NewGZIPDecompressor())}
+	if os.Getenv("GRPC_INSECURE") == "" {
+		// TODO: read from config
+		ethKey, err := ethcrypto.GenerateKey()
+		if err != nil {
+			cancel()
+			return nil, err
+		}
+		// Generate Certificate with signed publick key
+		certX509, priv, err := util.GenerateCert(ethKey)
+		if err != nil {
+			return nil, err
+		}
+		var cert = tls.Certificate{
+			Certificate: [][]byte{certX509.Raw},
+			PrivateKey:  priv,
+		}
+		servOpts = append(servOpts, grpc.Creds(util.NewTLS(&tls.Config{Certificates: []tls.Certificate{cert}})))
+	}
+	grpcServer := grpc.NewServer(servOpts...)
```

Comment:
Seems like there are options that are duplicated, like `RPCCompressor` etc here and in the `utils.MakeGrpcServer`. Maybe reuse?

Reply:
Gonna reuse, yes. Good idea.